packages/backend/src/managers/playground.ts

```diff
@@ -205,6 +207,26 @@ export class PlayGroundManager {
       }
     });
 
+    let contacted = false;
+    const start = Date.now();
+    while (
+      Date.now() - start < STARTING_TIME_MAX &&
+      !contacted &&
+      this.playgrounds.get(modelId).status === 'starting'
+    ) {
+      try {
+        await fetch(`http://localhost:${freePort}`);
+        contacted = true;
```

Comment:
```suggestion
        contacted = response.ok;
```

Reply:
nit: There are some use cases when response has error codes that does not trigger exceptions and host would be still unresponsive (network error in the 4xx–5xx range for example).